drivers/dma/dma_stm32_v3.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2025 Savoir-faire Linux, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef DMA_STM32_V3_H_
+#define DMA_STM32_V3_H_
+
+#include <stm32_ll_dma.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/drivers/dma.h>
+
+struct dma_stm32_descriptor {
+    uint32_t channel_tr1;
+    uint32_t channel_tr2;
+    uint32_t channel_br1;
+    uint32_t channel_sar;
+    uint32_t channel_dar;
+    uint32_t channel_llr;
+};
+
+struct dma_stm32_channel {
+	uint32_t direction;
+	bool hal_override;
+	volatile bool busy;
+	uint32_t state;
+	uint32_t src_size;
+	uint32_t dst_size;
+	void *user_data;
+	dma_callback_t dma_callback;
+	bool cyclic;
+	int block_count;
+};
+
+struct dma_stm32_data {
+	struct dma_context dma_ctx;
+};
+
+struct dma_stm32_config {
+	struct stm32_pclken pclken;
+	void (*config_irq)(const struct device *dev);
+	bool support_m2m;
+	uint32_t base;
```

Comment:
```suggestion
	DMA_TypeDef *base;
```
and remove all casts in driver